demo/mindstorms_widgets.py

```diff
@@ -0,0 +1,321 @@
+            else:
+                m.speed_regulation_enabled = 'off'
+                m.duty_cycle_sp = p
+                
+    def _move_wait( self ):
+        # wait for any running motors to finish their movement then stop
+        # all the motors.
+        left_is_off = False
+        right_is_off = False
+        if 'running' not in self.motor['left'].state:
+            left_is_off = True
+        if 'running' not in self.motor['right'].state:
+            right_is_off = True
+        while ((left_is_off or 'running' in self.motor['left'].state) and
+               (right_is_off or 'running' in self.motor['right'].state)):
+            time.sleep(0.1)
+        self.motor['left'].stop()
+        self.motor['right'].stop()
+        
+    def _move( self, mode, seconds, degrees, rotations ):
+        if mode == 'off':
+            self.motor['left'].stop()
+            self.motor['right'].stop()
+        elif mode == 'on':
+            self.motor['left'].run_forever()
+            self.motor['right'].run_forever()
+        elif mode == 'on_for_seconds':
+            self.motor['left'].run_forever()
+            self.motor['right'].run_forever()
+            time.sleep(seconds)
+            self.motor['left'].stop()
+            self.motor['right'].stop()
+        elif mode == 'on_for_degrees':
+            self.motor['left'].run_to_rel_pos(position_sp=degrees)
+            self.motor['right'].run_to_rel_pos(position_sp=degrees)
+            self._move_wait()
+        elif mode == 'on_for_rotations':
+            self.motor['left'].run_to_rel_pos( position_sp=int(rotations*360) )
+            self.motor['right'].run_to_rel_pos( position_sp=int(rotations*360) )
+            self._move_wait()
+            
+    def _validate_inputs( self, degrees, rotations ):
+        if (degrees < 0 or rotations < 0):
+            print( 'degrees and rotations should be positive. '
+                   'Use negative power to go backward' )
+
+    def _run_motor( self, motor, mode, power, 
+                    seconds, degrees, rotations,
+                    invert, regulated, brake_at_end ):
+        self._validate_inputs( degrees, rotations )
+        self._set_motor_attribs( 'medium', [motor], [power], invert, regulated,
+                                 brake_at_end )
+        if mode == 'off':
+            motor.stop()
+        elif mode == 'on':
+            motor.run_forever()
+        elif mode == 'on_for_seconds':
+            motor.run_forever()
+            time.sleep(seconds)
+            motor.stop()
+        elif mode == 'on_for_degrees':
+            motor.run_to_rel_pos(position_sp=degrees)
+            while 'running' in motor.state:
+                time.sleep(0.1)
+        elif mode == 'on_for_rotations':
+            motor.run_to_rel_pos(position_sp=int(rotations*360))
+            while 'running' in motor.state:
+                time.sleep(0.1)
+
+    # Begin public methods
+        
+    def connect_motor( self, motor ):

```

Comment:
Why not add a third parameter here, `port=None`, and stick to the default behavior when its not provided? This way we could allow non-standard connections as well.